### PR TITLE
Xenoarcheology removal

### PR DESCRIPTION
**Pull request by sgtmind**
First pass of the xenoarchealogy removal. Am leaving behind a couple items which have use outside of this.

## About The Pull Request

The removal of xenoarchaeology and all its components in order to make way for something better, or at least with far less spaghetti code. (Only reason I am doing this a second time is because I couldn't update my branch)

## Why It's Good For The Game

Honestly, had an awful experience with a random artifact spawning once, and I realized that xenoarcheaology adds nothing to the game. In addition, it is a bunch of spaghetti code

## Changelog
:cl:
del: Removed xenoarcheology
/:cl:

### CODE REVIEW

**Comment by ValoTheValo** [comment]
same as before. fuck xenoarch

**Comment by sgtmind** [comment]
What I am likely going to require help with is removing the part where it drops the artifact from mining.

**Comment by sgtmind** [comment]
How would I deal with the error of M.GetDrilled not being fouund?

File: code/modules/mining/ore.dm [hunk 1]
```diff
@@ -113,11 +113,4 @@
 /obj/item/ore/Initialize(mapload)
 	. = ..()
 	pixel_x = rand(0,16)-8
-	pixel_y = rand(0,8)-8
```
Comment (SirRichardFrancis) [inline code/modules/mining/ore.dm:116]: Maybe keep that one.

**Review by SirRichardFrancis — CHANGES_REQUESTED** [review]
You certainly have to keep `/turf/simulated/mineral`, but with most of the code removed.
GetDrilled() proc should stay, it's the one that makes ore drop.
Look where it's called, and make sure those calls are preserved, otherwise miners may have a problem.
Also please keep `/obj/structure/boulder` as a placeholder object, I'll remove it at a later date. Removal now would create annoying merge conflict.
`/datum/talking_atom` also should be kept, but moved elsewhere.
Location of `/obj/item/talkingcrystal` would be a good place for it.

**Comment by sgtmind** [comment]
Should be good, did a test, ore seems to drop, no artifacts within the code as far as I can see.

**Review by SirRichardFrancis — COMMENTED** [review]
`/obj/structure/boulder` and "excavation" should go too.

File: code/modules/mining/ore.dm [hunk 2]
```diff
@@ -113,11 +113,4 @@
...
-
-/obj/item/ore/attackby(obj/item/W, mob/user)
-	if(istype(W,/obj/item/device/core_sampler))
-		var/obj/item/device/core_sampler/C = W
-		C.sample_item(src, user)
-	else
-		return ..()
+	pixel_y = rand(0,8)-8
```
Comment (hyperioo) [inline code/modules/mining/ore.dm:116]:
```suggestion
	pixel_y = rand(0,8)-8

```
Newline

File: code/modules/reagents/prefilled.dm [hunk 3]
```diff
@@ -44,4 +44,4 @@
 	preloaded_reagents = list("fuel" = 60)
 
 /obj/item/reagent_containers/glass/beaker/silicon
-	preloaded_reagents = list("silicon" = 60)
+	preloaded_reagents = list("silicon" = 60)
```
Comment (hyperioo) [inline code/modules/reagents/prefilled.dm:47]:
```suggestion
	preloaded_reagents = list("silicon" = 60)

```
Newline

File: code/modules/mining/ore.dm (hunk 1 above)
Comment (hyperioo) [reply code/modules/mining/ore.dm:116]: It is kept just a few lines under

**Comment by hyperioo** [comment]
> `/obj/structure/boulder` and "excavation" should go too.

Why? Boulder seems to be still use when stuff is being excavated for ores

**Comment by SirRichardFrancis** [comment]
> Why? Boulder seems to be still use when stuff is being excavated for ores

And what purpose does it serve?

**Comment by sgtmind** [comment]
I am still not sure how to fix it

**Comment by SirRichardFrancis** [comment]
It's no big deal, could just let it be for now.

**Review by SirRichardFrancis — APPROVED** [review]

**Review by hyperioo — APPROVED** [review]

**Comment by vode-code** [comment]
> It's no big deal, could just let it be for now.

false, the result is the production of invulnerable boulders.

**Pull request merged**